prov/efa/test/efa_unit_test_ep.c

```diff
@@ -116,9 +116,12 @@ void test_efa_rdm_ep_handshake_exchange_host_id(struct efa_resource **state, uin
 	efa_rdm_ep = container_of(resource->ep, struct efa_rdm_ep, base_ep.util_ep.ep_fid);
 	efa_rdm_ep->host_id = g_efa_unit_test_mocks.local_host_id;
 	/* close shm_ep to force efa_rdm_ep to use efa device to send */
-	ret = fi_close(&efa_rdm_ep->shm_ep->fid);
-	assert_int_equal(ret, 0);
-	efa_rdm_ep->shm_ep = NULL;
+	if (efa_rdm_ep->shm_ep) {
```

Comment:
did we disable shm in this test? how come it wasn't broken?

Reply:
my env had the env set, and I figured this is a nice change to have.

Reply:
I can move this to a separate commit, or remove it if you want me to.

Reply:
moved to separate commit, and got other cases.